File: src/main/java/com/camackenzie/exvi/core/EnergyUnit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core;

/**
 *
 * @author callum
 */
public enum EnergyUnit implements Unit {
    KILOJOULE(1),
    KILOCALORIE(0.239006);

    private final double unit;

    private EnergyUnit(double u) {
        this.unit = u;
    }

    @Override
    public double getBaseCoefficient() {
        return this.unit;
    }

}

File: src/commonMain/kotlin/com/camackenzie/exvi/core/model/EnergyUnit.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.model

/**
 *
 * @author callum
 */
@kotlinx.serialization.Serializable
enum class EnergyUnit(private val unit: Double) : Unit {
    KILOJOULE(1.0), KILOCALORIE(0.239006);

    override fun getBaseCoefficient(): Double {
        return unit
    }
}